python/astro_metadata_translator/translator.py

```diff
@@ -1047,6 +1047,41 @@ def observing_date_to_offset(cls, observing_date: astropy.time.Time) -> astropy.
         """
         return None
 
+    @classmethod
+    def observing_date_to_observing_day(
+        cls, observing_date: astropy.time.Time, offset: astropy.time.TimeDelta | int | None
+    ) -> int:
+        """Return the YYYYMMDD integer corresponding to the observing day.
+
+        The offset is subtracted from the time of observation before
+        calculating the year, month and day.
+
+        Parameters
+        ----------
+        observing_date : `astropy.time.Time`
+            The observation date.
+        offset : `astropy.time.TimeDelta` | int | None
+            The offset to subtract from the observing date when calculating
+            the observing day. If `None` no offset is applied.
+
+        Returns
+        -------
+        day : `int`
+            The observing day as an integer of form YYYYMMDD.
+
+        Notes
+        -----
+        For example, if the offset is +12 hours both 2023-07-06T13:00 and
+        2023-07-07T11:00 will return an observing day of 20230706 because
+        the observing day goes from 2023-07-06T12:00 to 2023-07-07T12:00.
```

Comment:
Just curious - if the observing day starts one second before midnight, it is still attributed to the previous day?

Reply:
Yes, well, for Rubin that is how it works. For DECam we are defining day_obs ourselves.